inst/include/mode/mode.hpp

```diff
@@ -23,10 +23,9 @@ class container {
     // support multiple particles, and following the general approach
     // in dust will be sensible, I think.
     auto m = model_type(pars);
-    auto y = m.initial(time);
     auto ctl = control();
     for (size_t i = 0; i < n_particles; ++i) {
-      solver_.push_back(solver<model_type>(m, time, y, ctl));
+      solver_.push_back(solver<model_type>(m, time, ctl));
```

Comment:
I moved the initialisation of the y vector into the solver as it seemed to make more sense there

Reply:
A downside of this approach though is that we compute the initial conditions (incl the allocation) for every particle on reset

Reply:
Ah yeah, ok have changed this around again. One slightly odd thing is that now the container, the solver and the stepper all have a private model property because they all need to reference it in some way (to get initial conditions) which feels a bit weird, like there isn't a clear separation of concerns.

Reply:
If we made `initial_step_size` a method on the stepper we could entirely drop the model object from the solver at least. It can't be dropped from the stepper for sure (it *definitely* needs it). Dropping the model from the container object seems like it would be ideal - and we could do that as long as solver (via stepper) are capable of providing a vector of initial conditions.

This might be more obvious how to solve after we add the manual state setting though